src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture
+                .supplyAsync(
+                        () -> cache.computeIfAbsent(userId, this::getUserData),
+                        executor
+                );
```

Comment:
Using CompletableFuture with a cache is a good idea for asynchronous operations, but ensure that the computeIfAbsent method's performance implications are considered. It can potentially block other threads from accessing the cache if a long-running computation is in progress for a given key.